pkg/orchestrator/util.go

```diff
@@ -169,6 +170,51 @@ func getTelemetryTags(n NodeType) []string {
 	}
 }
 
+// CollectorType return collector type.
+func (n NodeType) CollectorType() int {
```

Comment:
Yes I plan to use the collector type that we've defined in the payload in backend to check the resource kind. https://github.com/DataDog/agent-payload/blob/master/process/message.go#L86-L113

Or we can extract resource kind field from the yaml in the backend, it's more dynamic

Reply:
hmm I don't have a big opinion on that part. It feels like we duplicate the logic everywhere, especially since we know the type. But we do need to have that information in the payload itself. Decoding the yaml first to get that information is too expensive.